tooling/greybox_fuzzer/src/mutation/mod.rs

```diff
@@ -0,0 +1,769 @@
+                    .map(|((name, typ), weight_node)| {
+                        (
+                            name.clone(),
+                            if mutation_weight >= weight_node.start
+                                && mutation_weight < weight_node.end
+                            {
+                                Self::splice_import_single_input_value(
+                                    typ,
+                                    &first_input_struct[name],
+                                    &second_input_struct[name],
+                                    weight_node,
+                                    mutation_weight,
+                                )
+                            } else {
+                                first_input_struct[name].clone()
+                            },
+                        )
+                    })
+                    .collect();
+
+                let fields: BTreeMap<_, _> = fields.into_iter().collect();
+                InputValue::Struct(fields)
+            }
+
+            AbiType::Tuple { fields } => {
+                let first_input_vector = match first_input {
+                    InputValue::Vec(previous_input_vector) => previous_input_vector,
+                    _ => panic!("Mismatch of AbiType and InputValue should not happen"),
+                };
+                let second_input_vector = match second_input {
+                    InputValue::Vec(previous_input_vector) => previous_input_vector,
+                    _ => panic!("Mismatch of AbiType and InputValue should not happen"),
+                };
+                let fields: Vec<_> = zip(fields, first_input_vector)
+                    .zip(second_input_vector)
+                    .zip(weight_tree_node.subnodes.as_ref().unwrap())
+                    .map(|(((typ, first_tuple_input), second_tuple_input), weight_node)| {
+                        if mutation_weight >= weight_node.start && mutation_weight < weight_node.end
+                        {
+                            Self::splice_import_single_input_value(
+                                typ,
+                                first_tuple_input,
+                                second_tuple_input,
+                                weight_node,
+                                mutation_weight,
+                            )
+                        } else {
+                            first_tuple_input.clone()
+                        }
+                    })
+                    .collect();
+                InputValue::Vec(fields)
+            }
+        }
+    }
+
+    /// Use one of splicing mechanisms to produce a new testcase generated by merging two previous ones
+    pub fn splice_two_maps(
+        &self,
+        first_input_map: &InputMap,
+        second_input_map: &InputMap,
+        prng: &mut XorShiftRng,
+    ) -> InputMap {
+        // TODO: add the most basic splice where the whole map is converted into a witness map and then we splice those
```

Comment:
Maybe capture a follow-up issue for this (or drop the TODO if it's not needed anymore)